### PR TITLE
FixCallSetSampleOrdering: support repairing callsets in which the sample in the file headers did not match the samples given to GenomicsDBImport

### PR DESCRIPTION
**Pull request by droazen**
Included a test case generated using the buggy version of GenomicsDBImport in which the sample names declared in the file headers did not always match the samples provided to GenomicsDBImport via the sample name map file, and showed that the tool could repair the callset successfully.

### CODE REVIEW

**Comment by droazen** [comment]
@eitanbanks please review (by testing out the repair tool)

**Comment by droazen** [comment]
Example usage:

```
./gatk-launch FixCallSetSampleOrdering 
-V src/test/resources/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering/badlysorted1000-batchSize50-readerThreads4-samplesInHeaderAreDifferent.vcf 
--sampleNameMap src/test/resources/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering/reverseOrderedWithHeaderMismatches.sample_map 
--batchSize 50 
--readerThreads 4 
--gvcfToHeaderSampleMapFile src/test/resources/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering/reverseOrderedWithHeaderMismatches.gvcfToHeaderSampleMapFile -O repaired.vcf
```

Note that in addition to the original sample name map file and batch size passed to GenomicsDBImport, you must also specify the original value of the --readerThreads argument. If this value was > 1, you are required to pass in a `--gvcfToHeaderSampleMapFile ` file giving the mapping from GVCF to sample name as declared in the GVCF header, one per line, separated by whitespace. The GVCFs in this file must match the GVCFs from the original sample name map file.

**Comment by droazen** [comment]
@lbergelson As discussed in person, I added an additional test case with a duplicated sample name in different batches, and the tests still passed.

**Comment by droazen** [comment]
@lbergelson Given that this fix has been proved to work, can we get a quick review/merge on this?

File: src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java [hunk 1]
```diff
@@ -91,13 +121,136 @@ public void onTraversalStart() {
...
+                    .collect(Collectors.toList());
+        }
+        else {
+            final List<String> batchSortedSamples = new ArrayList<>();
+
+            for ( final List<String> batch : batches ) {
+                // We have to compute these lookup tables per-batch, to allow for the possibility
+                // of duplicate sample names across different batches (but not within the same batch!)
+                Map<String, String> sampleNameInMapToSampleNameInHeader = new HashMap<>();
+                Map<String, String> sampleNameInHeaderToSampleNameInMap = new HashMap<>();
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java]: I still prefer a BiMap, but not worth changing at this point.

**Pull request merged**